tests/infra/remote.py

```diff
@@ -153,6 +153,7 @@ def __init__(
         self.out = os.path.join(self.root, "out")
         self.err = os.path.join(self.root, "err")
         self.suspension_proc = None
+        self.pid_file = "cchost.pid" if "./cchost" in self.cmd else "perf_client.pid"
```

Comment:
Can we make this more general? Something like `self.pid_file = f"{os.path.basename(self.cmd[0])}.pid}"` here, matched by `files::dump(fmt::format("{}.pid", argv[0]), ::getpid())` in both instances in C++? We can keep the `--pid-file` argument for overrides, by constructing its default from `argv[0]`.

Reply:
Good point. I've implemented this now.